### ISSUE TITLE
Installation : issue at step 5 when load dataset

### ISSUE THREAD

**Issue by bnouyou** [macarthur-lab/seqr#1060]
Hello,
I have just installed locally seqr, following all the steps. And now I am stuck at the step 5.
When I run the command :
`python2.7 gcloud_dataproc/submit.py --run-locally hail_scripts/v01/load_dataset_to_es.py  --spark-home $SPARK_HOME --genome-version $GENOME_VERSION --project-guid $PROJECT_GUID --sample-type $SAMPLE_TYPE --dataset-type $DATASET_TYPE --skip-validation  --exclude-hgmd --vep-block-size 100 --es-block-size 10 --num-shards 1 --hail-version 0.1 --use-nested-objects-for-vep --use-nested-objects-for-genotypes $INPUT_VCF`

I get this error at stage 7 :

`[Stage 7:====================================================>  (138 + 2) / 144]Summary(samples=2504, variants=145, call_rate=1.000000, contigs=['1'], multiallelics=0, snps=136, mnps=0, insertions=5, deletions=4, complex=0, star=0, max_alleles=2)
2019-10-29 15:15:05,863 INFO
==> total variants: 145`

`Can't locate DBI.pm in @INC (you may need to install the DBI module) (@INC contains: /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor /vep/loftee /usr/local/lib/perl5/site_perl/5.20.3/x86_64-linux /usr/local/lib/perl5/site_perl/5.20.3 /usr/local/lib/perl5/5.20.3/x86_64-linux /usr/local/lib/perl5/5.20.3 .) at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBConnection.pm line 75.
BEGIN failed--compilation aborted at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBConnection.pm line 75.
Compilation failed in require at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBAdaptor.pm line 69.
BEGIN failed--compilation aborted at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBAdaptor.pm line 69.
Compilation failed in require at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/Registry.pm line 137.
BEGIN failed--compilation aborted at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/Registry.pm line 137.
Compilation failed in require at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/Variation/Utils/VEP.pm line 80.
BEGIN failed--compilation aborted at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/Variation/Utils/VEP.pm line 80.
Compilation failed in require at /vep/variant_effect_predictor/variant_effect_predictor.pl line 72.
BEGIN failed--compilation aborted at /vep/variant_effect_predictor/variant_effect_predictor.pl line 72.`

I suppose this is an issue with perl, but I don't know how to solve it.

Thank you for your help.

**Comment by bnouyou** [comment]
Hello,
I have run this script and I have this error :
`mv: cannot stat ‘/etc/boto.cfg’: No such file or directory
`

**Comment by hanars** [comment]
Try re-running 
`SCRIPT=install_general_dependencies.sh && curl -L http://raw.githubusercontent.com/macarthur-lab/seqr/master/deploy/$SCRIPT -o $SCRIPT && chmod 777 $SCRIPT && ./$SCRIPT`
and then if that succeeds run the pipeline installation again

**Comment by bnouyou** [comment]
Hello,
I have re-run this script and I have the same error:
`===== init gsutil =====`
`Activated service account credentials for: [service-account@upheld-rookery-214823.iam.gserviceaccount.com]`
`mv: cannot stat ‘/etc/boto.cfg’: No such file or directory
`

**Comment by bnouyou** [comment]
Hello,
I have rerun the script `install_general_dependencies.sh`
I only have the error about the file `boto.cfg`
And then all the other scripts.

Now I am trying to load again the VCF file, and I still have the same error:

`[Stage 10:====================================================>(999 + 1) / 1000]Summary(samples=1, variants=129653, call_rate=0.973398, contigs=['X', '12', '8', '19', '4', '15', '11', '9', 'Y', '22', '13', '16', '5', '10', '21', '6', '1', '17', '14', '20', '2', '18', '7', '3'], multiallelics=0, snps=112264, mnps=2115, insertions=6622, deletions=8005, complex=647, star=0, max_alleles=2)
2019-11-04 16:15:11,181 INFO`
`==> total variants: 129653`
`Can't locate DBI.pm in @INC (you may need to install the DBI module) (@INC contains: /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor /vep/loftee /usr/local/lib/perl5/site_perl/5.20.3/x86_64-linux /usr/local/lib/perl5/site_perl/5.20.3 /usr/local/lib/perl5/5.20.3/x86_64-linux /usr/local/lib/perl5/5.20.3 .) at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBConnection.pm line 75.
BEGIN failed--compilation aborted at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBConnection.pm line 75.
Compilation failed in require at /seqr/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBAdaptor.pm line 69.`

**Comment by bw2** [comment]
@bnouyou VEP depends on DBI.pm (https://useast.ensembl.org/info/docs/tools/vep/script/vep_download.html#requirements), and perl isn't finding this module. 

The seqr pedigree drawing script also depends on it, so DBI is installed
https://github.com/macarthur-lab/seqr/blob/master/deploy/install_local.step8.install_pipeline_runner.sh#L56
and again here
https://github.com/macarthur-lab/seqr/blob/master/deploy/install_local.step6.install_seqr.sh#L75-L82

Can you please check your logs to see if those steps failed, or if the PERL5LIB is set in a way that prevents perl from finding this module?

**Comment by bnouyou** [comment]
Hello,

Thank you for your answer.

I have rerun the: https://github.com/macarthur-lab/seqr/blob/master/deploy/install_local.step8.install_pipeline_runner.sh#L56 
And I have: `DBI is up to date. (1.642) `

And then: https://github.com/macarthur-lab/seqr/blob/master/deploy/install_local.step6.install_seqr.sh#L75-L82
And here I have an error: 
`2019-11-05 09:31:08 URL:https://raw.githubusercontent.com/miyagawa/cpanminus/master/cpanm [302780/302780] -> "cpanm" [1]
--> Working on Cairo
Fetching http://www.cpan.org/authors/id/X/XA/XAOC/Cairo-1.107.tar.gz ... OK
==> Found dependencies: ExtUtils::MakeMaker, ExtUtils::Depends, ExtUtils::PkgConfig
--> Working on ExtUtils::MakeMaker
Fetching http://www.cpan.org/authors/id/B/BI/BINGOS/ExtUtils-MakeMaker-7.38.tar.gz ... OK
Configuring ExtUtils-MakeMaker-7.38 ... OK
Can't locate ExtUtils/Manifest.pm in @INC (@INC contains: FatPacked::35539848=HASH(0x21e4b88) /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at ./cpanm line 132.`


Moreover, I don't find PERL5LIB, how can I set it ? 

Thank you for your help.

**Comment by bnouyou** [comment]
Hello,

I have retried the installation from the beggining.
Now I have another issue, when installing redis.

In the logfile I have this error message :

`82579:C 21 Nov 2019 15:54:47.577 # Fatal error, can't open config file '/deploy/docker/redis/redis.conf'`

**Comment by Toseph** [comment]
export PERL5LIB=/your/perl/lib/path can be used to set your PERL5LIB to a non-standard perl library path.

**Comment by bnouyou** [comment]
Hello,
Thank you for your answer.
Actually I have re-started the installaton from the begining.
I found something quite strange in the step 6 :
For PLATFORM = centos, there is: 
` sudo apt-get update \
        && curl --silent --location https://rpm.nodesource.com/setup_8.x | sudo bash - \
        && sudo yum -y install nodejs`
apt-get should not be used with centos I suppose ?

**Comment by bnouyou** [comment]
Hello,

I have re-installed everything from the begining. And I still have the same issue with DBI.
See the error message:
`[Stage 10:====================================================>(994 + 4) / 1000]Summary(samples=1, variants=129653, call_rate=0.973398, contigs=['X', '12', '8', '19', '4', '15', '11', '9', 'Y', '22', '13', '16', '5', '10', '21', '6', '1', '17', '14', '20', '2', '18', '7', '3'], multiallelics=0, snps=112264, mnps=2115, insertions=6622, deletions=8005, complex=647, star=0, max_alleles=2)
2019-11-25 15:36:51,311 INFO
==> total variants: 129653
Can't locate DBI.pm in @INC (you may need to install the DBI module) (@INC contains: /seqr/softwares/SEQR/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor /vep/loftee /usr/local/lib/perl5/site_perl/5.20.3/x86_64-linux /usr/local/lib/perl5/site_perl/5.20.3 /usr/local/lib/perl5/5.20.3/x86_64-linux /usr/local/lib/perl5/5.20.3 .) at /seqr/softwares/SEQR/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBConnection.pm line 75.
BEGIN failed--compilation aborted at /seqr/softwares/SEQR/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBConnection.pm line 75.
Compilation failed in require at /seqr/softwares/SEQR/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBAdaptor.pm line 69.
BEGIN failed--compilation aborted at /seqr/softwares/SEQR/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/DBSQL/DBAdaptor.pm line 69.
Compilation failed in require at /seqr/softwares/SEQR/seqr/vep/ensembl-tools-release-85/scripts/variant_effect_predictor/Bio/EnsEMBL/Registry.pm line 137.
`

I use platform CentOS.

Thank you in advance for your help.

**Comment by hanars** [comment]
Can you double check that after the full re-install DBI is installed and PERL5LIB is set?

**Comment by bnouyou** [comment]
At which step should it be set ?
After the full re-install, PERL5LIB is empty for me.

**Comment by hanars** [comment]
It should be set after step 6, and then step 8 should confirm that it is set
